hibernate-core/src/main/java/org/hibernate/type/TypeFactory.java

```diff
@@ -131,12 +131,13 @@ public Type type(Class<Type> typeClass, Properties parameters) {
 	}
 
 	// todo : can a Properties be wrapped in unmodifiable in any way?
-	private final static Properties EMPTY_PROPERTIES = new Properties();
+	private final static ConcurrentHashMapBackedProperties EMPTY_PROPERTIES = new ConcurrentHashMapBackedProperties();
```

Comment:
Why not using a `ConcurrentHashMap<String, String>` instead?

Reply:
@vladmihalcea Added additional commit to use ConcurrentHashMap<String,String> in place of ConcurrentHashMap<Object,Object>. Since properties are of type Hashtable and always contain String types we can simply typecast to Map.